src/modulus_convert.rs

```diff
@@ -0,0 +1,148 @@
+use crate::error::BoxError;
+use crate::field::Field;
+use crate::helpers::ring::Ring;
+use crate::prss::Participant;
+use crate::replicated_secret_sharing::ReplicatedSecretSharing;
+use crate::securemul::{ProtocolContext, SecureMul};
+use bit_vec::BitVec;
+
+pub enum HelperIdentity {
+    H1,
+    H2,
+    H3,
+}
+
+pub struct RandomShareGenerationHelper<'a> {
+    rng: &'a Participant,
+    identity: HelperIdentity,
+}
+
+impl<'a> RandomShareGenerationHelper<'a> {
+    #[must_use]
+    pub fn new(rng: &'a Participant, identity: HelperIdentity) -> Self {
+        Self { rng, identity }
```

Comment:
I think that maybe we need to revise how Participant is created and used.  What we really want here is the ability to create multiple streams of shared state from a single key exchange.  I'll take a look at this today.